### PR TITLE
allow dev comments to be enabled from cli, opt in

### LINKED ISSUE #132: make operations on dev comments opt-in
**Describe the bug**

Current master always includes developer comments.

**To Reproduce**

Use the default settings to `reflow` or `check` and observe
dev comments being operated on, which is a breaking change.

**Expected behavior**

Dev comments being checked _optionally_ (cli flag and/or config option).

Ref #115

### PR DESCRIPTION
**Pull request by drahnr**
## What does this PR accomplish?

 * 🩹 Bug Fix
 * 🦚 Feature

Closes #132 .

## Changes proposed by this PR:

A minor simplification attempt on cli args, and threads a `bool` to the places that need it.

## 📜 Checklist

 * [x] Works on the `./demo` sub directory
 * [x] Test coverage is excellent and passes
 * [x] Documentation is thorough

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -118,6 +132,27 @@ struct Args {
...
+impl Args {
+    fn action(&self) -> Action {
+        // extract operation mode
+        let action = if self.cmd_fix || self.flag_fix {
+            Action::Fix
+        } else if self.cmd_reflow {
+            Action::Reflow
+        } else if self.cmd_config {
+            Action::Config
+        } else if self.flag_help {
```
Comment (KuabeM) [inline src/main.rs:144]: It seems to me that this could lead to `--help` not being recognised fi `fix` is present.

Comment (KuabeM) [reply src/main.rs:144]: Nevermind, docopt does not allow both flags

File: src/main.rs [hunk 2]
```diff
@@ -70,6 +69,19 @@ Options:
   --skip-readme             Do not attempt to process README.md files listed in Cargo.toml manifests.
 "#;
 
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, Deserialize)]
+enum CheckerType {
+    #[serde(alias = "hunspell")]
```
Comment (KuabeM) [inline src/main.rs:74]:
```suggestion
    #[serde(alias = "hunspell")]
    #[serde(alias = "Hunspell")]
```

**Review by KuabeM — APPROVED** [review]
Passing the bool `dev_comments` around seems not too ergonomic, but it's the easiest way to integrate this fast and without too much refactoring :+1:

File: src/main.rs (hunk 1 above)
Comment (drahnr) [reply src/main.rs:144]: Exactly :) and this is necessary for the case where `cargo-spellcheck` is invoked via `cargo spellcheck` without further arguments.

File: src/main.rs (hunk 2 above)
Comment (drahnr) [reply src/main.rs:74]: It's already named `Hunspell`, so that is already covered iiu the serde docs correctly.

**Comment by drahnr** [comment]
I agree it's not pretty, especially having it in `traverse` is just wrong from the logic partitioning. Needs a refactor, but I want to get this feature out the door with the next alpha, and refactor during the alpha / beta releases.

**Pull request merged**